src/arith/solve_linear_equation.cc

```
@@ -0,0 +1,476 @@
+    PrimExpr e = te::make_zero(system_to_solve->variables[i].dtype());
+    for (size_t j = 0; j < num_vars; ++j) {
+      e = e + te::make_const(e.dtype(), V[i][j])*solution_for_V_inv_x[j];
+    }
+    e = analyzer_problem.Simplify(e);
+    old_to_new_map.Set(system_to_solve->variables[i], e);
+  }
+
+  // The resulting ranges
+  Map<Var, Range> new_ranges = InferRange(
+      new_to_old_map, system_to_solve->variables, system_to_solve->ranges);
+  Analyzer analyzer_solution;
+  analyzer_solution.Bind(new_ranges);
+
+  // We have to transform ranges of the old variables into relations over new variables because
+  // new ranges are not enough usually.
+  for (const auto& p : system_to_solve->ranges) {
+    const Var& old_var = p.first;
+    const Range& old_range = p.second;
+    if (old_to_new_map.count(old_var)) {
+      PrimExpr express_by_new_vars = old_to_new_map[old_var];
+      PrimExpr lower_cond = analyzer_solution.Simplify(
+          old_range->min <= express_by_new_vars);
+      PrimExpr upper_cond = analyzer_solution.Simplify(
+          express_by_new_vars < old_range->min + old_range->extent);
+      if (!tir::is_const_int(lower_cond, 1)) {
+        new_relations.push_back(lower_cond);
+      }
+      if (!tir::is_const_int(upper_cond, 1)) {
+        new_relations.push_back(upper_cond);
+      }
+    }
+  }
+
+  // Add the rest conditions
+  for (const PrimExpr& cond : rest) {
+    new_relations.push_back(Substitute(cond, old_to_new_map));
+  }
+
+  LinearSystem solution(new_vars, new_ranges, new_relations);
+  LinearSystemTransform transform(
+      system_to_solve, solution, old_to_new_map, new_to_old_map);
+
+  return transform;
+}
+
+TVM_REGISTER_GLOBAL("arith.SolveEquations")
+.set_body([](TVMArgs args, TVMRetValue *ret) {
+    if (args.size() == 1) {
+      *ret = SolveEquations(args[0]);
+    } else if (args.size() == 3) {
+      LinearSystem problem(args[0], args[1], args[2]);
+      *ret = SolveEquations(problem);
+    }
```

should fail explicitly for args of other size